cmd/lndinit/cmd_migrate_db.go

```diff
@@ -0,0 +1,595 @@
+		log("Copying top-level bucket '%s'", loggableKeyName(key))
+
+		destTx, err := destDb.BeginReadWriteTx()
+		if err != nil {
+			return err
+		}
+
+		destBucket, err := destTx.CreateTopLevelBucket(key)
+		if err != nil {
+			return fmt.Errorf("error creating top level bucket "+
+				"'%s': %v", loggableKeyName(key), err)
+		}
+
+		srcBucket := srcTx.ReadWriteBucket(key)
+		err = copyBucketKvdb(srcBucket, destBucket)
+		if err != nil {
+			return fmt.Errorf("error copying bucket '%s': %v",
+				loggableKeyName(key), err)
+		}
+
+		log("Committing bucket '%s'", loggableKeyName(key))
+		if err := destTx.Commit(); err != nil {
+			return fmt.Errorf("error committing bucket '%s': %v",
+				loggableKeyName(key), err)
+		}
+
+		return nil
+
+	})
+	if err != nil {
+		return fmt.Errorf("error enumerating top level buckets: %v",
+			err)
+	}
+
+	// Migrate wallet created marker.
+	if prefix == lncfg.NSWalletDB && x.Dest.isRemote() {
+		const (
+			walletMetaBucket = "lnwallet"
+			walletReadyKey   = "ready"
+		)
+
+		log("Creating 'wallet created' marker")
+		destTx, err := destDb.BeginReadWriteTx()
+		if err != nil {
+			return err
+		}
+
+		metaBucket, err := destTx.CreateTopLevelBucket(
+			[]byte(walletMetaBucket),
+		)
+		if err != nil {
+			return err
+		}
+
+		err = metaBucket.Put(
+			[]byte(walletReadyKey), []byte(walletReadyKey),
+		)
+		if err != nil {
+			return err
+		}
+
+		log("Committing 'wallet created' marker")
+		if err := destTx.Commit(); err != nil {
+			return fmt.Errorf("error committing 'wallet created' "+
+				"marker: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func copyBucketKvdb(src walletdb.ReadWriteBucket,
+	dest walletdb.ReadWriteBucket) error {
+
+	if err := dest.SetSequence(src.Sequence()); err != nil {
+		return fmt.Errorf("error copying sequence number")
+	}
+
+	return src.ForEach(func(k, v []byte) error {
+		if v == nil {
+			srcBucket := src.NestedReadWriteBucket(k)
+			destBucket, err := dest.CreateBucket(k)
+			if err != nil {
+				return fmt.Errorf("error creating bucket "+
+					"'%s': %v", loggableKeyName(k), err)
+			}
+
+			if err := copyBucketKvdb(srcBucket, destBucket); err != nil {
+				return fmt.Errorf("error copying bucket "+
+					"'%s': %v", loggableKeyName(k), err)
+			}
+
+			return nil
+		}
+
+		err := dest.Put(k, v)
+		if err != nil {
+			return fmt.Errorf("error copying key '%s': %v",
+				loggableKeyName(k), err)
+		}
+
+		return nil
+	})
+}
+
+func (x *migrateDBCommand) migrateEtcd(srcTx walletdb.ReadWriteTx,
+	prefix string) error {
+
+	ctx := context.Background()
+	cfg := x.Dest.Etcd.CloneWithSubNamespace(prefix)
+	destDb, ctx, cancel, err := etcd.NewEtcdClient(ctx, *cfg)
+	if err != nil {
+		return err
+	}
+	defer cancel()
+
+	err = srcTx.ForEachBucket(func(key []byte) error {
+		log("Copying top-level bucket '%s'", loggableKeyName(key))
+
+		return migrateBucketEtcd(
+			ctx, destDb, []string{string(key)},
+			srcTx.ReadWriteBucket(key),
+		)
+	})
+	if err != nil {
+		return err
```

Comment:
Not sure but perhaps we could delete all keys at this point? Just to avoid leaving any junk in the DB.